File: C plus/1000.cpp
#include <bits/stdc++.h>
#include <iostream>
#include <string>
using namespace std;
int main(){
    int l,n;
    bool check = false;
    int count = 0;
    string first;
    string word,result;
    cin >> l ;
    cin >> n ;
    cin >> first;
    if (n==1){
        cout << first << endl;
    }else{
        while(count < n-1){
            int tar = 0;
            cin >> word;
            for (int i = 0;i <l;i++){
                if (first[i] != word[i]){
                    tar++;
                }
            }
            count++;
            if (tar > 2 && !check){
                result = first;
                check = true;
            }
            first = word;
        }
        if (check){
            cout << result << endl;
        }else{
            cout << word << endl;
        }
    }
    return 0;
}

File: Chapter 1/1000.py
l = int(input())
n = int(input())
check = False
count = 0
first = input()
if n ==1:
    print(first)
else:
    while count < n-1:
        tar = 0
        word = input()
        for i in range(l):
            if first[i] == word[i]:
                pass
            else:
                tar += 1
        count +=1
        if tar > 2 and not check:
            result = first
            check = True
        first = word
    if check:   
        print(result)
    else:
        print(word)

